### DISCUSSION TITLE
How use reaction with flow?

### DISCUSSION THREAD

**Discussion by Yrsafam** [mobxjs/mobx-state-tree#1919] [Q&A] [answered]
Hi! In my case, I need to make a request to the server when changing the field. The data acquisition function is implemented using flow. When I try to call a function in reaction, I get an error: 

> a mst flow must always have a parent context

If you redo the method of receiving data on async \await, then everything works as it should. But I would like to use flow as recommended by the library. I assume that in the case of flow, there is a loss of this context. Unfortunately, I could not solve this problem on my own.
Sample code below:

```js
const fetchData = flow(function* () {
      // ...
      let result = yield getData();
      // ...
});

function afterCreate() {
      const disposerReaction = reaction(
        () => self.field,
        flow(function* () {
          yield fetchData ();
        })
      );
      
      addDisposer(self, disposerReaction)
}
```

in the case when the store is not a descendant of another, then this problem can be solved by a hack in the form of:
```js
function afterCreate() {
      const disposerReaction = reaction(
        () => self.field,
        flow(function* () {
          yield getRoot(self).fetchData();
        })
      );
      
      addDisposer(self, disposerReaction)
}
```

**Comment by EmilTholin** [comment, marked as answer]
Hi @Yrsafam!

You could probably put `fetchData` in a separate `actions` block to have it always have a parent context:

```js
// ...
.actions((self) => ({
  fetchData: flow(function* () {
    // ...
    let result = yield getData();
    // ...
  }),
}))
.actions((self) => ({
  afterCreate() {
    const disposerReaction = reaction(
      () => self.field,
      self.fetchData
    );

    addDisposer(self, disposerReaction);
  },
}));
```

**Reply by Yrsafam** [reply]
It's brilliant! Thanks!
